File: CPP_Joc_Windows_Android/SH_Client_Win_Cpp_Cmake/App/src/rpg3D/gw/entity/template/TeamTemplate.cpp
#include <rpg3D/gw/entity/module/TeamModule.h>
#include "TeamTemplate.h"

using namespace rpg3D;

TeamTemplate::TeamTemplate()
	: super("team", ITeamModule::TYPE)
{
	//void
}

IEntityModule* TeamTemplate::createNewModuleIfExists(IWorldEntity* entity, BasicModuleTemplate* _t, std::shared_ptr<ExtraData> extraData, IEntityModuleCreator* moduleCreator) {
	TeamTemplate* t = dynamic_cast<TeamTemplate*>(_t);
	if(!t) {
		throw LogicException(LOC);
	}

	TeamModule* mod = new TeamModule(entity, t);

	return mod;
}

TeamTemplate::~TeamTemplate() {
	//void
}

File: NodeJs_TypeScript_JavaScript_Server/Project_NodeJs_VSCode/src/squareheads/adventlandia/basiczone/gwi/entity/template/TeamTemplate.ts
import { BasicModuleTemplate } from "localbase/gwi/entity/template/BasicModuleTemplate";

export class TeamTemplate extends BasicModuleTemplate {
    
  public static readonly MODULE_ID: string = "team";
  
  public constructor() {
    super(TeamTemplate.MODULE_ID, true);
    
    //void
  }

}
